### PR TITLE
Fix customlabel

### PR DESCRIPTION
**Pull request by ZeHiro**
[Review on Reviewable](https://reviewable.io/reviews/bnjbvr/kresus/311)

### CODE REVIEW

**Comment by bnjbvr** [comment]
Wow, this code is hard to read. Let's try to make it simpler and comment it a bit.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[client/components/OperationList.js, line 65 \[r1\]](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-K76gBLHaC0PPNy_iU5q:-K76gBLHaC0PPNy_iU5r:1969825066) ([raw file](https://github.com/bnjbvr/kresus/blob/2a22e188d2c070a11a40e7fd40643d25bc583a7c/client/components/OperationList.js#L65)):*
Okay, these conditions are really hard to read. Let's restructure them and add comments:

```
if (customLabel) {
    // If the new non empty customLabel value is different from the current one, save it.
    if (customLabel !== this.defaultValue() &&
        customLabel.trim().length) {
        code();
    }
} else if (this.props.operations.customLabel &&
                this.props.operations.customLabel.length) {
    // If the new customLabel value is empty and there was already one, unset it.
    code();
}
```

Also, to make it clearer, maybe rename `customLabel` to `newCustomLabel`?

---

*[client/components/OperationList.js, line 77 \[r1\]](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-K76h-jdA9ubA8hIAYbi:-K76h-jdA9ubA8hIAYbj:175194043) ([raw file](https://github.com/bnjbvr/kresus/blob/2a22e188d2c070a11a40e7fd40643d25bc583a7c/client/components/OperationList.js#L77)):*
I think we'll want helpers at some point, so if you feel motivated, please put the 13 and 27 constants into helpers.js in a plain old object called KEYS, such that KEYS.Escape === 27 and KEYS.Enter === 13. I'll do it in a future commit otherwise.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-:-K76hJN7MeMUXWhmWmP6:-151047408)*

**Comment by nicofrand** [comment]
According to https://facebook.github.io/react/docs/events.html and https://facebook.github.io/react/blog/2014/07/17/react-v0.11.html#improved-keyboard-event-normalization, `event.key` could be used instead of `event.keyCode` with the value equal to one of the values defined in [the DOM Level 3 Events keys](http://www.w3.org/TR/DOM-Level-3-Events-key/).

**Comment by bnjbvr** [comment]
Sounds good to me.

**Comment by ZeHiro** [comment]
@bnjbvr good to merge for me.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-:-K78McmuSgufo-3PjmxJ:686790512)*

**Comment by bnjbvr** [comment]
Looks good, testing and merging.

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-:-K7Aj7y4SK3xK6n2HhcZ:292426644)*

File: client/components/OperationList.js [hunk 1]
```diff
@@ -56,24 +56,27 @@ class LabelComponent extends React.Component {
 
     onBlur() {
         let customLabel = this.dom().value;
-        if (customLabel && customLabel !== this.defaultValue()) {
-            if (customLabel.trim().length) {
+        if (customLabel) {
+            // If the new non empty customLabel value is different from the current one, save it.
+            if (customLabel !== this.defaultValue() && customLabel.trim().length) {
```
Comment (bnjbvr) [inline client/components/OperationList.js]: Sorry for getting back to this, but I realized we could do slightly better, so as not to save customLabels which are the default value plus a few spaces:

`if (customLabel.trim().length && customLabel.trim() !== this.defaultValue())`

**Comment by ZeHiro** [comment]
@bnjbvr, ready to merge. 

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/311#-:-K7DZuSYtze4JltLMn01:402379377)*

**Comment by bnjbvr** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/311)*

**Comment by bnjbvr** [comment]
There was a typo in the first commit's message + the content was wrongly dispatched among commits, but I didn't want to bother you with all of this again (this PR has been quite long already), so I've amended them and pushed afterward (note that you're still marked as the author of the patch). Closing as your commits have merged as 6b945c217d5617afdf5c0aef7c653c5642fef0f7 and 713c105b2610d79fc3c90a0f5113cbe9ac15f852 .

**Comment by ZeHiro** [comment]
No worry. As long as it is merged

**Pull request closed without merging**